src/main/java/org/skriptlang/reflect/java/elements/structures/StructImport.java

```diff
@@ -34,76 +31,66 @@
 import java.util.regex.Pattern;
 import java.util.stream.StreamSupport;
 
-public class CustomImport {
+public class StructImport extends Structure {
+
+  public static final Priority PRIORITY = new Priority(150);
   private static final Pattern IMPORT_STATEMENT =
       Pattern.compile("(" + SkriptMirrorUtil.PACKAGE + ")(?:\\s+as (" + SkriptMirrorUtil.IDENTIFIER + "))?");
+  private static final SyntaxElementInfo<?> thisInfo;
+  private static final Map<Script, Map<String, JavaType>> imports = new HashMap<>();
 
   static {
-    CustomSyntaxSection.register("Import", SectionImport.class, "import");
+    Skript.registerStructure(StructImport.class, "import");
     Skript.registerEffect(EffImport.class, "import <" + IMPORT_STATEMENT.pattern() + ">");
 
     // TODO try replacing ImportHandler with JavaType's literal parsing
     Skript.registerExpression(ImportHandler.class, JavaType.class, ExpressionType.SIMPLE);
-    Optional<ExpressionInfo<?, ?>> info = StreamSupport.stream(
-        Spliterators.spliteratorUnknownSize(Skript.getExpressions(), Spliterator.ORDERED), false)
-        .filter(i -> i.c == ImportHandler.class)
-        .findFirst();
-
-    if (info.isPresent()) {
-      thisInfo = info.get();
-    } else {
-      Skript.warning("Could not find custom import class. Custom imports will not work.");
-    }
+    thisInfo = StreamSupport.stream(Spliterators.spliteratorUnknownSize(Skript.getExpressions(), Spliterator.ORDERED), false)
+        .filter(expressionInfo -> ImportHandler.class.equals(expressionInfo.getElementClass()))
+        .findFirst().orElseGet(() -> {
+          Skript.warning("Could not find custom import class. Custom imports will not work.");
```

Comment:
Would this ever be the case? Why do we not do this check for the other custom element classes?

Reply:
I'm not sure. Should I remove it from here or add it to the custom element classes?

Reply:
IMO either remove it or add a comment like "TODO figure out why this is here" 🤣